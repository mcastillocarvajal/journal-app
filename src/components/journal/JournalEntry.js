import React from 'react'

export const JournalEntry = () => {
    return (
        <div className="journal__entry pointer">

            <div 
                className="journal__entry-picture"
                style={{
                    backgroundSize: 'cover',
                    backgroundImage: 'url(https://imagekit.io/blog/content/images/2019/12/image-optimization.jpg)'
                }}
            ></div>

            <div className="journal__entry-body">

                <p className="journal__entry-title">
                    First day
                </p>
                <p className="journal__entry-content">
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                </p>
            </div>

            <div className="journal__entry-date-box">
                <span>Monday</span>
                <h4>25</h4>
            </div>

        </div>
    )
}
