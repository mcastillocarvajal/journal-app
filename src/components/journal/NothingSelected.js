import React from 'react'

export const NothingSelected = () => {
    return (
        <div className="nothing__main-content">

            <p>
                Select a note or
                <br />
                create a new entry
            </p>

            <i className="far fa-star fa-4x mt-5"/>

        </div>
    )
}
