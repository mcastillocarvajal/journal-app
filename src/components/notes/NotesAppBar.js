import React from 'react'

export const NotesAppBar = () => {
    return (
        <div className="notes__appbar">
            <span>January 1, 2012</span>

            <div>

                <button className="btn">
                    Picture
                </button>

                <button className="btn">
                    Save
                </button>

            </div>
        </div>
    )
}
