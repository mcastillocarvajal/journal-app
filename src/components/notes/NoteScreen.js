import React, { useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { activeNote } from '../../actions/notes';
import { useForm } from '../../hooks/useForm';
import { NotesAppBar } from './NotesAppBar'


export const NoteScreen = () => {

    const dispatch = useDispatch();

    const { active:note } = useSelector(state => state.notes);

    const [ formValues, handleInputChange, reset ] = useForm( note );

    const { body, title } = formValues;

    const activeId = useRef( note.id )

    useEffect(() => {

        if ( note.id !== activeId.current ) {
            reset( note );
            activeId.current = note.id
        }

    }, [ note, reset ]);

    useEffect(() => {

        dispatch( activeNote( formValues.id, { ...formValues} ) );

    }, [ formValues, dispatch ])


    return (
        <div className="notes__main-content">

            <NotesAppBar />

            <div className="notes__content">

                <input  
                    type="text"
                    className="notes__title-input"
                    placeholder="Some awesome words"
                    autoComplete="off"
                    name='title'
                    value={ title }
                    onChange={ handleInputChange }
                />

                <textarea 
                    placeholder="What happened today?"
                    className="notes__textarea"
                    name='body'
                    value={ body }
                    onChange={ handleInputChange }
                ></textarea>
            </div>

            {
                (note.url) &&
                <div className="notes-image">

                    <img
                        src="https://www.businessinsider.in/photo/81769906/How-to-reverse-image-search-on-Google-to-find-information-related-to-a-specific-photo.jpg?imgsize=297676"
                        alt="img"
                    />
                </div>
            }

        </div>
    )
}
