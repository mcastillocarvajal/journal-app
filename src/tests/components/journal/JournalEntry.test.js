import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore( initState );
store.dispatch = jest.fn();

const note = {
    id: '1234',
    title: 'title',
    body: 'body',
    url: 'url',
    date: 0
}

const wrapper = mount( 
    <Provider store={ store }>
            <JournalEntry { ...note } />
    </Provider>
);

describe('test for <JournalEntry />', () => {
    
    test('should match snapshot', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should trigger activeNote', () => {
        
        wrapper.find('.journal__entry').prop('onClick')();

        expect( store.dispatch ).toHaveBeenCalledWith(
            activeNote( note.id, { ...note } )
        );
    })
    
})
