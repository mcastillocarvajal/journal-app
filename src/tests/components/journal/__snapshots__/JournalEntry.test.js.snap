// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for <JournalEntry /> should match snapshot 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(url)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      title
    </p>
    <p
      className="journal__entry-content"
    >
      body
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
       
      Wednesday
       
    </span>
    <h4>
       
      31st
       
    </h4>
  </div>
</div>
`;
